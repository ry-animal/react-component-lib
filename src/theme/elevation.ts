export const ELEVATION_LIGHT = {
  low: "0px 2px 8px RGBA(0, 0, 0, 0.04)",
  medium: "0px 2px 16px RGBA(0, 0, 0, 0.16)",
  high: "0px 2px 24px RGBA(0, 0, 0, 0.24)",
};

export const ELEVATION_DARK = {
  low: "inherit",
  medium: "inherit",
  high: "inherit",
};
