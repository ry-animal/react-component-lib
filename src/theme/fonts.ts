import "@fontsource/open-sans/400.css";
import "@fontsource/open-sans/500.css";
import "@fontsource/open-sans/600.css";
import "@fontsource/open-sans/700.css";
import "@fontsource/poppins/400.css";
import "@fontsource/poppins/500.css";
import "@fontsource/poppins/600.css";
import "@fontsource/poppins/700.css";

export const FontSize = {
  XXXXL: "72px",
  XXXL: "56px",
  XXL: "40px",
  XL: "32px",
  LG: "24px",
  MD: "20px",
  SM: "18px",
  XS: "16px",
  XXS: "14px",
  XXXS: "12px",
};

/*
 * https://developer.mozilla.org/en-US/docs/Web/CSS/font-weight
 * 400 is regular, 500 is medium, 700 is bold
 */
export const FontWeight = {
  NORMAL: 400,
  MEDIUM: 500,
  SEMI_BOLD: 600,
  BOLD: 700,
};

export const FONT = {
  family: {
    title: "'Poppins', sans-serif",
    body: "'Poppins', sans-serif",
  },
  size: {
    ...FontSize,
  },
  weight: {
    ...FontWeight,
  },
};

export const FontFamily = {
  SANS_SERIF_TITLE: FONT.family.title,
  SANS_SERIF_BODY: FONT.family.body,
};
