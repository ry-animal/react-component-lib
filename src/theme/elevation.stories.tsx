import styled from "styled-components";
import { Text } from "../components/Text";
import { ELEVATION_LIGHT } from "./elevation";

const BoxContainer = styled.div`
  display: flex;
  padding: ${({ theme }) => theme.space.s24};
  flex-direction: row;
  margin-bottom: ${({ theme }) => theme.space.s8};
`;

interface BoxProps {
  item: string;
}
const Box = styled.div<BoxProps>`
  align-self: center;
  background-color: ${({ theme }) => theme.color.SURFACE_2};
  margin-right: ${({ theme }) => theme.space.s32};
  width: 300px;
  height: 300px;
  box-shadow: ${({ theme, item }) => {
    if (item === "low") {
      return theme.elevation.low;
    }
    if (item === "high") {
      return theme.elevation.high;
    }
    return theme.elevation.medium;
  }};
`;

export default {
  title: "Theme/Elevation",
  parameters: { options: { showPanel: false } },
};

const ElevationItem = ({ name, value }: { name: string; value: string }) => (
  <BoxContainer>
    <Box item={name} />
    <div>
      <Text>
        <b>Key:</b> {name}
      </Text>
      <Text>
        <b>Value:</b> {value}
      </Text>
    </div>
  </BoxContainer>
);

export const ElevationList = (): JSX.Element[] =>
  Object.entries(ELEVATION_LIGHT).map(([key, value]) => (
    <ElevationItem key={key} name={key} value={value} />
  ));
