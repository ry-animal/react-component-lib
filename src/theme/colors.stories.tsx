/* eslint-disable no-magic-numbers */
import * as React from "react";
import { withDesign } from "storybook-addon-designs";
import { Switch, Text } from "../";
import { PALETTE_DARK, PALETTE_LIGHT } from "./colors";

export default {
  title: "Theme/Colors",
  decorators: [withDesign],
};

const colorBlock = {
  marginBottom: 32,
} as React.CSSProperties;

const Item = ({
  itemKey,
  itemValue,
  color,
}: {
  itemKey: string;
  itemValue: string;
  color: string;
}) => {
  const renderStyles = {
    margin: 0,
    padding: "32px 16px 16px",
    backgroundColor: itemValue,
  } as React.CSSProperties;

  return (
    <div>
      <div style={renderStyles}>
        <Text text={itemKey} color={color} weight="700" reset />
        <Text
          text={itemValue}
          variant="Caption"
          style={{
            color,
            textTransform: "uppercase",
          }}
          reset
        />
      </div>
    </div>
  );
};

const getKey = (position: number) => Object.keys(PALETTE_LIGHT)[position];
const getColors = (isDark: boolean) => (isDark ? PALETTE_DARK : PALETTE_LIGHT);

interface ColorGroup {
  isDark?: boolean;
}

function Primary({ isDark = false }: ColorGroup) {
  return (
    <div style={colorBlock}>
      <Text
        text="Primary"
        variant="Display4"
        color={getColors(isDark).TEXT_1}
      />
      <Item
        itemKey={getKey(0)}
        itemValue={getColors(isDark).PRIMARY_1}
        color={isDark ? "black" : "white"}
      />
      <Item
        itemKey={getKey(1)}
        itemValue={getColors(isDark).PRIMARY_2}
        color={isDark ? "black" : "white"}
      />
      <Item
        itemKey={getKey(2)}
        itemValue={getColors(isDark).PRIMARY_3}
        color={isDark ? "white" : "black"}
      />
    </div>
  );
}

function Secondary({ isDark = false }: ColorGroup) {
  return (
    <div style={colorBlock}>
      <Text
        text="Secondary"
        variant="Display4"
        color={getColors(isDark).TEXT_1}
      />
      <Item
        itemKey={getKey(3)}
        itemValue={getColors(isDark).SECONDARY_1}
        color="black"
      />
      <Item
        itemKey={getKey(4)}
        itemValue={getColors(isDark).SECONDARY_2}
        color="black"
      />
      <Item
        itemKey={getKey(5)}
        itemValue={getColors(isDark).SECONDARY_3}
        color="black"
      />
    </div>
  );
}

function Success({ isDark = false }: ColorGroup) {
  return (
    <div style={colorBlock}>
      <Text
        text="Success"
        variant="Display4"
        color={getColors(isDark).TEXT_1}
      />
      <Item
        itemKey={getKey(6)}
        itemValue={getColors(isDark).SUCCESS_1}
        color={isDark ? "white" : "black"}
      />
      <Item
        itemKey={getKey(7)}
        itemValue={getColors(isDark).SUCCESS_2}
        color="white"
      />
      <Item
        itemKey={getKey(8)}
        itemValue={getColors(isDark).SUCCESS_3}
        color="white"
      />
    </div>
  );
}

function Error({ isDark = false }: ColorGroup) {
  return (
    <div style={colorBlock}>
      <Text text="Error" variant="Display4" color={getColors(isDark).TEXT_1} />
      <Item
        itemKey={getKey(9)}
        itemValue={getColors(isDark).ERROR_1}
        color={isDark ? "white" : "black"}
      />
      <Item
        itemKey={getKey(10)}
        itemValue={getColors(isDark).ERROR_2}
        color="white"
      />
      <Item
        itemKey={getKey(11)}
        itemValue={getColors(isDark).ERROR_3}
        color="white"
      />
    </div>
  );
}

function Surface({ isDark = false }: ColorGroup) {
  return (
    <div style={colorBlock}>
      <Text
        text="Surface"
        variant="Display4"
        color={getColors(isDark).TEXT_1}
      />
      <Item
        itemKey={getKey(12)}
        itemValue={getColors(isDark).SURFACE_1}
        color={isDark ? "white" : "black"}
      />
      <Item
        itemKey={getKey(13)}
        itemValue={getColors(isDark).SURFACE_2}
        color={isDark ? "white" : "black"}
      />
      <Item
        itemKey={getKey(14)}
        itemValue={getColors(isDark).SURFACE_3}
        color={isDark ? "white" : "black"}
      />
      <Item
        itemKey={getKey(15)}
        itemValue={getColors(isDark).SURFACE_4}
        color={isDark ? "white" : "black"}
      />
    </div>
  );
}

function Texto({ isDark = false }: ColorGroup) {
  return (
    <div style={colorBlock}>
      <Text text="Text" variant="Display4" color={getColors(isDark).TEXT_1} />
      <Item
        itemKey={getKey(16)}
        itemValue={getColors(isDark).TEXT_1}
        color={isDark ? "black" : "white"}
      />
      <Item
        itemKey={getKey(17)}
        itemValue={getColors(isDark).TEXT_2}
        color={isDark ? "black" : "white"}
      />
      <Item
        itemKey={getKey(18)}
        itemValue={getColors(isDark).TEXT_3}
        color={isDark ? "black" : "white"}
      />
      <Item
        itemKey={getKey(19)}
        itemValue={getColors(isDark).TEXT_4}
        color={isDark ? "black" : "white"}
      />
    </div>
  );
}

function Basic({ isDark = false }: ColorGroup) {
  return (
    <div style={colorBlock}>
      <Text text="Basic" variant="Display4" color={getColors(isDark).TEXT_1} />
      <Item
        itemKey={getKey(20)}
        itemValue={getColors(isDark).WHITE}
        color="black"
      />
      <Item
        itemKey={getKey(21)}
        itemValue={getColors(isDark).BLACK}
        color="white"
      />
    </div>
  );
}

export const LightThemePalette = () => (
  <div
    style={{
      backgroundColor: PALETTE_LIGHT.SURFACE_1,
      margin: -16,
      padding: 16,
    }}
  >
    <Primary />
    <Secondary />
    <Success />
    <Error />
    <Surface />
    <Texto />
    <Basic />
  </div>
);
LightThemePalette.parameters = {
  design: {
    type: "figma",
    url: "https://www.figma.com/file/YzgT8FhyEaHw5s1rimOnkn/NFT-Marketplace-v1?node-id=10847%3A6766",
  },
};

export const DarkThemePalette = () => (
  <div
    style={{
      backgroundColor: PALETTE_DARK.SURFACE_1,
      margin: -16,
      padding: 16,
    }}
  >
    <Primary isDark />
    <Secondary isDark />
    <Success isDark />
    <Error isDark />
    <Surface isDark />
    <Texto isDark />
    <Basic isDark />
  </div>
);

DarkThemePalette.parameters = {
  design: {
    type: "figma",
    url: "https://www.figma.com/file/YzgT8FhyEaHw5s1rimOnkn/NFT-Marketplace-v1?node-id=10847%3A6825",
  },
};

export const ThemeSwitcher = () => {
  const [isDark, setIsdark] = React.useState(false);
  const handleRadioChange = () => setIsdark(!isDark);

  return (
    <>
      <div style={colorBlock}>
        <Switch
          checked={isDark}
          label="Dark Mode"
          onChange={handleRadioChange}
        />
      </div>
      {isDark ? <DarkThemePalette /> : <LightThemePalette />}
    </>
  );
};
ThemeSwitcher.parameters = {
  design: {
    type: "figma",
    url: "https://www.figma.com/file/YzgT8FhyEaHw5s1rimOnkn/NFT-Marketplace-v1?node-id=10158%3A130155",
  },
};
