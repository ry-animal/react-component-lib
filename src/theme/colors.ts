import { ThemeColors, ThemeName } from "./theme.interface";

export const lightPaletteColors = {
  PRIMARY_1: "#000000",
  PRIMARY_2: "#5F5F5F",
  PRIMARY_3: "#EEEEEE",

  SECONDARY_1: "#FFF6DA",
  SECONDARY_2: "#FFE8A1",
  SECONDARY_3: "#FFD968",

  SUCCESS_1: "#F2FBF8",
  SUCCESS_2: "#00B176",
  SUCCESS_3: "#007D53",

  ERROR_1: "#FFF2F4",
  ERROR_2: "#BC2828",
  ERROR_3: "#710601",
};

export const lightPaletteBase = {
  SURFACE_1: "#F9F9F9",
  SURFACE_2: "#FFFFFF",
  SURFACE_3: "#EEEEEE",
  SURFACE_4: "#DBDBDB",

  TEXT_1: "#151515",
  TEXT_2: "#404040",
  TEXT_3: "#5F5F5F",
  TEXT_4: "#A4A4A4",
};

export const darkPaletteColors = {
  PRIMARY_1: "#F9F9F9",
  PRIMARY_2: "#EEEEEE",
  PRIMARY_3: "#151515",

  SECONDARY_1: "#FFE8A1",
  SECONDARY_2: "#FFD968",
  SECONDARY_3: "#B38402",

  SUCCESS_1: "#19C68C",
  SUCCESS_2: "#00B176",
  SUCCESS_3: "#007D53",

  ERROR_1: "#FFF2F4",
  ERROR_2: "#BC2828",
  ERROR_3: "#710601",
};

export const darkPaletteBase = {
  SURFACE_1: "#151515",
  SURFACE_2: "#212322",
  SURFACE_3: "#404040",
  SURFACE_4: "#5F5F5F",

  TEXT_1: "#FFFFFF",
  TEXT_2: "#EEEEEE",
  TEXT_3: "#DBDBDB",
  TEXT_4: "#C3C3C3",
};

export const palletteBase = {
  BLACK: "#000000",
  WHITE: "#ffffff",
};

export const PALETTE_LIGHT: ThemeColors = {
  ...lightPaletteColors,
  ...lightPaletteBase,
  ...palletteBase,
};

export const PALETTE_DARK: ThemeColors = {
  ...darkPaletteColors,
  ...darkPaletteBase,
  ...palletteBase,
};

export function getColorsByTheme(theme: ThemeName): ThemeColors {
  if (theme === ThemeName.DARK) {
    return PALETTE_DARK;
  }

  return PALETTE_LIGHT;
}
