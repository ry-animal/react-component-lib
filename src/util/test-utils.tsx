import { render, RenderOptions } from "@testing-library/react";
import "jest-styled-components";
import * as React from "react";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import { LIGHT_THEME } from "../theme";

interface ProvidersProps {
  children: React.ReactNode;
}

function Providers({ children }: ProvidersProps) {
  return (
    <BrowserRouter>
      <ThemeProvider theme={LIGHT_THEME}>{children}</ThemeProvider>
    </BrowserRouter>
  );
}

const customRender = (
  ui: JSX.Element,
  options?:
    | RenderOptions<
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement
      >
    | undefined
) => render(ui, { wrapper: Providers, ...options });

// Duplicated instead of import "../../__mocks__/matchMediaMock"; because jest has trouble resolving imports
Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

// Re-export everything
export * from "@testing-library/react";
// Override render method
export { customRender as render };
