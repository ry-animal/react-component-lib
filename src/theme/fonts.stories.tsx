import * as React from "react";
import { lightPaletteColors } from "./colors";
import { FONT } from "./fonts";

export default {
  title: "Theme/Fonts",
  parameters: { options: { showPanel: false } },
};

const codeStyles = {
  backgroundColor: lightPaletteColors.PRIMARY_3,
  padding: "2px 8px",
  marginBottom: "62px",
  fontSize: "14px",
  fontFamily: "Lucida Console, Courier, monospace",
} as React.CSSProperties;

const Item = ({
  itemKey,
  itemValue,
  fontType,
}: {
  itemKey: string;
  itemValue: string;
  fontType: string;
}) => {
  const renderStyles = {
    margin: 0,
    lineHeight: 1,
    fontFamily: fontType === "family" ? itemValue : FONT.family.title,
    fontSize: fontType === "size" ? itemValue : FONT.size.XXL,
    fontWeight: fontType === "weight" ? itemValue : FONT.weight.SEMI_BOLD,
  } as React.CSSProperties;

  return (
    <>
      <h3 style={renderStyles}>The quick brown fox jumps over the lazy dog</h3>
      <pre style={codeStyles}>
        Key: {itemKey} - Value: {itemValue}
      </pre>
    </>
  );
};

const mapFonts = (obj: object, fontType: string): React.ReactNode => {
  const fonts: React.ReactNodeArray = [];
  Object.entries(obj).forEach(([key, value]) =>
    fonts.push(
      <Item itemKey={key} itemValue={value} fontType={fontType} key={key} />
    )
  );

  return fonts;
};

export const Family = () => mapFonts(FONT.family, "family");

export const Size = () => mapFonts(FONT.size, "size");

export const Weight = () => mapFonts(FONT.weight, "weight");
