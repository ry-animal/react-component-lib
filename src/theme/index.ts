import { getColorsByTheme } from "./colors";
import { ELEVATION_DARK, ELEVATION_LIGHT } from "./elevation";
import { FONT, FontFamily, FontSize, FontWeight } from "./fonts";
import { GRADIENT } from "./gradient";
import { LAYERS } from "./layers";
import { MEDIA_WIDTHS } from "./media";
import { OPACITY } from "./opacity";
import { RADIUS } from "./radius";
import { SPACE } from "./space";
import { ThemeName } from "./theme.interface";

const sharedThemeKeys = {
  radius: {
    ...RADIUS,
  },
  space: {
    ...SPACE,
  },
  type: {
    ...FONT,
  },
  layer: {
    ...LAYERS,
  },
  media: {
    ...MEDIA_WIDTHS,
  },
  opacity: {
    ...OPACITY,
  },
  gradient: {
    ...GRADIENT,
  },
};

export const lightColors = getColorsByTheme(ThemeName.LIGHT);
export const darkColors = getColorsByTheme(ThemeName.DARK);

export const LIGHT_THEME = {
  name: ThemeName.LIGHT,
  ...sharedThemeKeys,
  color: {
    ...lightColors,
  },
  elevation: {
    ...ELEVATION_LIGHT,
  },
};

export const DARK_THEME = {
  name: ThemeName.DARK,
  ...sharedThemeKeys,
  color: {
    ...darkColors,
  },
  elevation: {
    ...ELEVATION_DARK,
  },
};

// Legacy
export { FontFamily, FontSize, FontWeight, MEDIA_WIDTHS, SPACE };
