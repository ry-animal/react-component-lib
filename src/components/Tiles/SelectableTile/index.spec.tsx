import userEvent from "@testing-library/user-event";
import { SelectableTile } from ".";
import { lightColors } from "../../../theme";
import { render, screen } from "../../../util/test-utils";

describe("<SelectableTile/>", () => {
  const sampleProps = () => ({
    title: "Collection name",
    subtitle: "15 items",
    onClick: jest.fn(),
  });

  describe("rendering", () => {
    test("default matches snapshot", () => {
      expect(render(<SelectableTile />).asFragment()).toMatchSnapshot();
    });

    test("tile with sample props matches snapshot", () => {
      expect(
        render(<SelectableTile {...sampleProps()} />).asFragment()
      ).toMatchSnapshot();
    });

    test("renders text elements", () => {
      const props = sampleProps();
      render(<SelectableTile title={props.title} subtitle={props.subtitle} />);
      expect(screen.getByText(props.title)).toBeInTheDocument();
      expect(screen.getByText(props.subtitle)).toBeInTheDocument();
    });

    test("renders selected tile style", () => {
      const props = sampleProps();
      render(<SelectableTile {...props} selected />);
      expect(screen.getByRole("button")).toHaveStyleRule(
        "background",
        lightColors.SURFACE_3
      );
    });

    test("renders img element", () => {
      const props = sampleProps();
      const imgUrl = "./imgPath.jpg";
      render(<SelectableTile {...props} iconUrl={imgUrl} />);
      expect(screen.getByAltText(`${props.title} icon`)).toHaveAttribute(
        "src",
        imgUrl
      );
    });

    test("renders custom display element", () => {
      const props = sampleProps();
      const title = "display-content";
      render(
        <SelectableTile {...props} displayContent={<div title={title} />} />
      );
      expect(screen.getByTitle(title)).toBeInTheDocument();
    });

    test("disabled button", () => {
      const props = sampleProps();
      render(<SelectableTile {...props} disabled />);
      expect(screen.getByRole("button")).toHaveStyleRule("opacity", "0.7", {
        modifier: "&:disabled",
      });
    });
  });

  describe("event handling", () => {
    test("tile can be clicked", () => {
      const props = sampleProps();
      render(<SelectableTile {...props} />);
      userEvent.click(screen.getByRole("button"));
      expect(props.onClick).toBeCalled();
    });

    test("disabled tile cannot be clicked", () => {
      const props = sampleProps();
      render(<SelectableTile {...props} disabled />);
      userEvent.click(screen.getByRole("button"));
      expect(props.onClick).not.toBeCalled();
    });
  });
});
