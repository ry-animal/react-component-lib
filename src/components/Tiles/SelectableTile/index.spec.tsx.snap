// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectableTile/> rendering default matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  cursor: pointer;
  -webkit-transition: -webkit-transform 0.25s ease-out,box-shadow 0.25s ease-out;
  -webkit-transition: transform 0.25s ease-out,box-shadow 0.25s ease-out;
  transition: transform 0.25s ease-out,box-shadow 0.25s ease-out;
  gap: 20px;
  border-radius: 8px;
  background: #F9F9F9;
  border: 2px solid transparent;
  padding: 32px;
  color: #151515;
}

.c0:hover {
  -webkit-transform: scale(1.01);
  -ms-transform: scale(1.01);
  transform: scale(1.01);
  box-shadow: 0px 2px 16px RGBA(0,0,0,0.16);
}

.c0:disabled {
  cursor: default;
  opacity: 0.7;
}

.c0:disabled:hover {
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  box-shadow: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<button
    class="c0"
  >
    <div
      class="c1"
    />
  </button>
</DocumentFragment>
`;

exports[`<SelectableTile/> rendering tile with sample props matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  cursor: pointer;
  -webkit-transition: -webkit-transform 0.25s ease-out,box-shadow 0.25s ease-out;
  -webkit-transition: transform 0.25s ease-out,box-shadow 0.25s ease-out;
  transition: transform 0.25s ease-out,box-shadow 0.25s ease-out;
  gap: 20px;
  border-radius: 8px;
  background: #F9F9F9;
  border: 2px solid transparent;
  padding: 32px;
  color: #151515;
}

.c0:hover {
  -webkit-transform: scale(1.01);
  -ms-transform: scale(1.01);
  transform: scale(1.01);
  box-shadow: 0px 2px 16px RGBA(0,0,0,0.16);
}

.c0:disabled {
  cursor: default;
  opacity: 0.7;
}

.c0:disabled:hover {
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  box-shadow: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  font-size: 16px;
  font-weight: 600;
  line-height: 1.25rem;
  text-align: center;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  word-break: break-word;
}

.c3 {
  text-align: center;
}

<button
    class="c0"
  >
    <div
      class="c1"
    >
      <span
        class="c2"
      >
        Collection name
      </span>
      <span
        class="c3"
      >
        15 items
      </span>
    </div>
  </button>
</DocumentFragment>
`;
